web/packages/core/src/ruffle-player.ts

```diff
@@ -667,6 +669,38 @@ export class RufflePlayer extends HTMLElement {
         }
     }
 
+    /**
+     * Async helper function for fetching the SWF.
+     */
+    private async fetchSWF(): Promise<void> {
+        const response = await fetch(this.swfUrl!);
+        if (!response.ok) {
+            console.error("SWF download failed");
+            return;
+        }
+        const blob = await response.blob();
+        const blobUrl = window.URL.createObjectURL(blob);
+        const swfDownloadA = document.createElement("a");
+        swfDownloadA.style.display = "none";
+        swfDownloadA.href = blobUrl;
+        swfDownloadA.download = this.swfUrl!.replace(/^.*\//, "");
+        document.body.appendChild(swfDownloadA);
+        swfDownloadA.click();
+        window.URL.revokeObjectURL(blobUrl);
```

Comment:
Let's reverse the following 2 lines to preserve a LIFO state (create URL, append to DOM, remove from DOM, revoke URL).